### DISCUSSION TITLE
Request to include modifyResponseBodyData as an available service method

### DISCUSSION THREAD

**Discussion by mattmilburn** [ComfortablyCoding/strapi-plugin-transformer#88] [Ideas]
Hello and thank you for all your work on this plugin! Also hope you are feeling better after being sick recently 🙂  (just noticed in issue comments)

I recently went to upgrade from a v2 version of the plugin to v3 and realized I was relying on the previous version of the plugin's `response` service method in one of my app's custom middlewares  but now it seems that is not possible to use directly from the plugin.

After reviewing the latest version of the plugin, do you think it would be possible to make the `modifyResponseBodyData` util function an available service method?

**Background and Example**
In my app, I have some `Pages` which can have parent pages and some have more than one parent page. For the API response data, I'm using a middleware which finds the **root parent page** and appends it to the response data, but it's using the old `response` method from the transformer plugin to sanitize the new data.

```js
const plugin = strapi.plugin( 'transformer' );
const settings = plugin.service( 'settingsService' ).get();
const transform = plugin.service( 'transformService' );

// Find the root page to append to the response data.
const rootPage = await strapi.query( 'api::page.page' ).findOne( {
  where: {
    slug: 'example-root-page',
  },
} );

// Format data so it uses data and attributes, then let the transformer plugin deal with it according to plugin settings.
const rootPageData = {
  data: {
    id: rootPage.id,
    attributes: omit( rootPage, [ 'id' ] ),
  },
};

// Final result.
const entityWithExtras = {
   ...entity, // assume the `entity` object is already defined.
   root_page: transform.response( settings, rootPage ),
};
```

**Comment by ComfortablyCoding** [comment]
Thanks!

Interesting use case. I did not expose these underlying functions in v3 due to them requiring a breaking change (and therefore a new major version) if i wanted to modify the function name, signature etc in v2.

I am not opposed  to exposing them under an exprimental property on the service with the understanding that the functions are not gaurunteed to be (but most likely will be) stable. 

What do you think?

**Reply by mattmilburn** [reply]
Hi @ComfortablyCoding Yes that sounds good to me 🙂  Sorry for the late reply!
